subsys/bluetooth/host/id.c

```diff
@@ -1802,8 +1811,6 @@ int bt_id_set_scan_own_addr(bool active_scan, uint8_t *own_addr_type)
 			return err;
 		}
 	} else {
-		*own_addr_type = bt_dev.id_addr[0].type;
```

Comment:
What should happen if `CONFIG_BT_SCAN_WITH_IDENTITY` is not enabled but `!is_adv_using_rand_addr())` is false? Prior to this change we would still set `*own_addr_type` here. Now `*own_addr_type` will be left undefined in this case.

Reply:
I think you right. 
With `CONFIG_BT_SCAN_WITH_IDENTITY=n` and `is_adv_using_rand_addr() == false`, then `*own_addr_type` is not assigned. 

I think the previous assignment, `*own_addr_type = bt_dev.id_addr[0].type;`, was incorrect, as `id_addr[0].type` and the `BT_HCI_OWN_ADDR*` were in 2 different value spaces. 

Since this may happen when `CONFIG_BT_SCAN_WITH_IDENTITY=n` and `CONFIG_BT_PRIVACY=n` should the resulting address type be `BT_HCI_OWN_ADDR_RANDOM`?

Reply:
I'm not sure why the code (and the original code) is guarded by `!is_adv_using_rand_addr()`, but it's old code it seems. @jhedberg @alwa-nordic any idea/suggestions?

Reply:
I think it would be good to move this discussion out of a closed PR into a new issue, to give it the attention it deserves. 

As for the background to this code, I can't say, at least not without doing some closer investigation into the matter.

Reply:
https://github.com/zephyrproject-rtos/zephyr/issues/85202